metronome/hotstuff/service/src/io/iohk/metronome/hotstuff/service/storage/BlockPruning.scala

```diff
@@ -0,0 +1,36 @@
+package io.iohk.metronome.hotstuff.service.storage
+
+import io.iohk.metronome.hotstuff.consensus.basic.Agreement
+import io.iohk.metronome.storage.KVStore
+
+import cats.implicits._
+
+object BlockPruning {
+
+  def prune[N, A <: Agreement](
+      blockStorage: BlockStorage[N, A],
+      viewStateStorage: ViewStateStorage[N, A],
+      blockHistorySize: Int
+  ): KVStore[N, Unit] = {
+    for {
+      // Always keep the last executed block.
+      lastExecutedBlock <- viewStateStorage.getLastExecutedBlockHash.lift
+      pathFromRoot      <- blockStorage.getPathFromRoot(lastExecutedBlock).lift
+
+      // Keep the last N blocks.
+      pruneable = pathFromRoot.reverse
```

Comment:
I believe the name is somewhat confusing, they're actually `nonPruneable`, aren't they?

Reply:
No these are definitely pruneable. Maybe the comment was confusing :thinking: 

I'm not sure why I haven't just used `pathFromRoot.dropRight(blockHistorySize)` here, but the point is that we can prune all but the last N items. I'm going to do that by calling `pruneNonDescendants` on the last item of the `pruneable` list.

Reply:
Tried to add more descriptive comments.